dvc/compare.py

```diff
@@ -124,10 +126,17 @@ def __len__(self) -> int:
     def shape(self) -> Tuple[int, int]:
         return len(self.columns), len(self)
 
-    def drop(self, *col_names: str) -> None:
-        for col_name in col_names:
-            self._keys.remove(col_name)
-            self._columns.pop(col_name)
+    def drop(self, *col_names: str, keep: Optional[List[str]] = None) -> None:
```

Comment:
Maybe it is better to split this into two functions? A pure drop or `__del__` and a `drop_column` it is quite weird for a `drop` function to have a `keep` option for me.

Reply:
Because I added support for fnmatch , It felt like there should be a way to have a keep.

Perhaps if instead of fnmatch we support regex we could remove keep argument

Reply:
So, it is a `drop` supporting `wild match` 🤔 . Another way is to add a new `protect` function to protect some of the columns and can set once and effect in `drop`, `dropna` and `drop_duplicate`.

Reply:
I like the `protect` idea

Reply:
Added `protected` property (get only) and `protect`/`unprotect` methods. Removed `keep` arg from `drop`

Reply:
ping @karajan1001 , could you review?